### PR TITLE
errors: add resource group throttled error

### LINKED ISSUE #41675: Slow failure when specified resource group doesn't exist
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

use test;
create table t1 (c1 int);
set global tidb_enable_resource_controkl='on';
select /+* resource_group(tg1) */ * from t1;

### 2. What did you expect to see? (Required)
ERROR 1105 (HY000): [resource group] resourceGroupName rg1 is not existed be thrown in seconds.

### 3. What did you see instead (Required)

The error code is thrown after ~50 seconds.

image

### 4. What is your TiDB version? (Required)

https://github.com/pingcap/tidb/pull/41678

### PR DESCRIPTION
**Pull request by CabinfeverB**
### What problem does this PR solve?

Issue Number: ref #38825 close #41675

Problem Summary:

### What is changed and how it works?
Some resource group errors are defined. And the resource group errors returned from KV will be converted at the kv layer.
After https://github.com/tikv/client-go/pull/722, we can update client-go to fix #41675.

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [x] Manual test (add detailed scripts or steps below)
image

- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by CabinfeverB** [comment]
cc @BornChanger @glorv @nolouch @JmPotato

File: store/driver/error/error.go [hunk 1]
```diff
@@ -53,6 +54,12 @@ var (
 	ErrPDServerTimeout = dbterror.ClassTiKV.NewStd(errno.ErrPDServerTimeout)
 	// ErrRegionUnavailable is the error when region is not available.
 	ErrRegionUnavailable = dbterror.ClassTiKV.NewStd(errno.ErrRegionUnavailable)
+	// ErrResourceGroupNotExists is the error when resource group does not exist.
+	ErrResourceGroupNotExists = dbterror.ClassTiKV.NewStd(errno.ErrResourceGroupNotExists)
+	// ErrResourceGroupConfigUnavailable is the error when resource group configuration is unavailable.
+	ErrResourceGroupConfigUnavailable = dbterror.ClassTiKV.NewStd(errno.ErrResourceGroupConfigUnavailable)
+	// ErrResourceGroupLimited is the error when
```
Comment (HuSharp) [inline store/driver/error/error.go:61]: seem like lost comment

**Review by nolouch — APPROVED** [review]
lgtm

**Review by glorv — APPROVED** [review]
LGTM

**Pull request merged**